what4/src/What4/Expr/Builder.hs

```diff
@@ -1091,6 +1091,19 @@ sbConcreteLookup sym arr0 mcidx idx
   | Just (ConstantArray _ _ v) <- asApp arr0 = do
       return v
 
+  | Just (UpdateArray range idx_tps arr update_idx v) <- asApp arr0
```

Comment:
Let me suggest that this is complex enough to warrant a high-level comment describing this case